pkg/kpimon/kpi.go

```diff
@@ -51,43 +54,78 @@ func runListMetricsCommand(cmd *cobra.Command, args []string) error {
 	writer := new(tabwriter.Writer)
 	writer.Init(outputWriter, 0, 0, 3, ' ', tabwriter.FilterHTML)
 
-	request := kpimonapi.GetRequest{
-		Id: "kpimon",
-	}
+	request := kpimonapi.GetRequest{}
 	client := kpimonapi.NewKpimonClient(conn)
-	respGetHeader, err := client.GetMetricTypes(context.Background(), &request)
-	if err != nil {
-		return err
-	}
-	respGetMetrics, err := client.GetMetrics(context.Background(), &request)
+
+	respGetMeasurement, err := client.GetMeasurement(context.Background(), &request)
 	if err != nil {
 		return err
 	}
 
-	for k, v := range respGetMetrics.GetObject().GetAttributes() {
-		ids := strings.Split(k, ":")
-		tmpCid := ids[0]
-		tmpPlmnID := ids[1]
-		tmpEgnbID := ids[2]
-		tmpMetricType := ids[3]
-		tmpTimestamp := ids[4]
-		tmpKey := fmt.Sprintf("%s:%s:%s", tmpCid, tmpPlmnID, tmpEgnbID)
-
-		if _, ok1 := results[tmpKey]; !ok1 {
-			results[tmpKey] = make(map[string]map[string]string)
-		}
-		if _, ok2 := results[tmpKey][tmpTimestamp]; !ok2 {
-			results[tmpKey][tmpTimestamp] = make(map[string]string)
+	attr := make(map[string]string)
+	for key, measItems := range respGetMeasurement.GetMeasurements() {
+		for _, measItem := range measItems.MeasurementItems {
+			for _, measRecord := range measItem.MeasurementRecords {
+				timeStamp := measRecord.Timestamp
+				measName := measRecord.MeasurementName
+				measValue := measRecord.MeasurementValue
+
+				if _, ok := attr[measName]; !ok {
+					attr[measName] = measName
+				}
+
+				if _, ok1 := results[key]; !ok1 {
+					results[key] = make(map[uint64]map[string]string)
+				}
+				if _, ok2 := results[key][timeStamp]; !ok2 {
+					results[key][timeStamp] = make(map[string]string)
+				}
+
+				var value interface{}
+
+				name, err := prototypes.AnyMessageName(measValue)
+				if err != nil {
+					return err
+				}
+
+				switch name {
+				case "onos.kpimon.IntegerValue":
```

Comment:
Perhaps you can get the message name from the actual message in generated code instead of hard coding it.

Reply:
It should .Name or something